translator/GCP_cloud_function/cloud_function/icone_translator.py

```diff
@@ -372,48 +407,17 @@ def initialize_info():
     info['metadata'] = {}
     info['metadata']['wz_location_method'] = "channel-device-method"
     info['metadata']['lrs_type'] = "lrs_type"
-    info['metadata']['location_verify_method'] = "location_verify_method"
-    # info['metadata']['datafeed_frequency_update'] = 86400
-    # info['metadata']['timestamp_metadata_update'] = "timestamp_metadata_update"
-    info['metadata']['contact_name'] = "Abinash Konersman"
+    info['metadata']['contact_name'] = "Abinash Konersman"  # we can consider to add a representive name from iCone
     info['metadata']['contact_email'] = "abinash.konersman@state.co.us"
-    info['metadata']['issuing_organization'] = "CDOT"
+    info['metadata']['issuing_organization'] = "iCone"
 
     return info
 
-
-def parse_xml(inputfile):
-    with open(inputfile, encoding='utf-8-sig') as frsm:
-        # Read
-        xmlSTRING = frsm.read()
-        icone_obj = xmltodict.parse(xmlSTRING)
-        return icone_obj
-
-
-def validate_write(wzdx_obj, outputfile, location_schema):
-    wzdx_schema = json.loads(open(location_schema).read())
-    try:
-        validate(instance=wzdx_obj, schema=wzdx_schema)
-    except ValidationError as e:
-        print(e)
-        return False
-
-    with open(outputfile, 'w') as fwzdx:
-        fwzdx.write(json.dumps(wzdx_obj, indent=2))
-
-    # if not os.path.exists(outputfile) or os.stat(outputfile).st_size == 0:
-    #     print('Output file creation failed . file does not exist or is empty')
+def validate_wzdx(wzdx_obj, wzdx_schema):
+    #wzdx_schema = json.loads(open(location_schema).read())
+    # try:
```

Comment:
Should this `try` still be in the file? If not and we come upon an invalid file, it will throw an error and cause an exception.

Reply:
we removed the commented code and we are erring on the side of only pushing valid data .if validation failed ,we want to throw an exception.

Reply:
So what happens when an exception is thrown? It doesn't look like we are handling it, or logging anything.

Reply:
Looks like https://cdot-its.atlassian.net/browse/CV-64?atlOrigin=eyJpIjoiMDA3OTAwZDFlYzNkNDJiZmFkMTQ5ZWJkYjBkMmIyNDQiLCJwIjoiaiJ9 will handle this. Closing